src/Umbraco.Core/Persistence/Repositories/ContentTypeRepository.cs

```diff
@@ -171,24 +171,37 @@ protected override void PersistNewItem(IContentType entity)
 
             ((ContentType)entity).AddingEntity();
 
-            var factory = new ContentTypeFactory(NodeObjectTypeId);
-            var dto = factory.BuildDto(entity);
+            PersistNewBaseContentType(entity);
+            PersisteTemplates(entity, false);
 
-            PersistNewBaseContentType(dto.ContentTypeDto, entity);
-            //Inserts data into the cmsDocumentType table if a template exists
-            if (dto.TemplateNodeId > 0)
+            entity.ResetDirtyProperties();
+        }
+
+        protected void PersisteTemplates(IContentType entity, bool clearAll)

```

Comment:
I've merged this branch locally and already fixed this up, so no need for this one any more!